serde_v8/src/magic/buffer.rs

```diff
@@ -14,6 +14,8 @@ pub enum MagicBuffer {
   ToV8(Cell<Option<Box<[u8]>>>),
 }
 
+unsafe impl Sync for MagicBuffer {}
```

Comment:
If a Mutex instead of a Cell is used, MagicBuffer will automatically inherit Sync.
```diff
diff --git a/serde_v8/src/magic/buffer.rs b/serde_v8/src/magic/buffer.rs
index c14284c7..7ffff861 100644
--- a/serde_v8/src/magic/buffer.rs
+++ b/serde_v8/src/magic/buffer.rs
@@ -1,9 +1,9 @@
 use rusty_v8 as v8;
 
-use std::cell::Cell;
 use std::fmt;
 use std::ops::Deref;
 use std::ops::DerefMut;
+use std::sync::Mutex;
 
 use super::zero_copy_buf::ZeroCopyBuf;
 
@@ -11,11 +11,9 @@ use super::zero_copy_buf::ZeroCopyBuf;
 // allowing us to use a single type for familiarity
 pub enum MagicBuffer {
   FromV8(ZeroCopyBuf),
-  ToV8(Cell<Option<Box<[u8]>>>),
+  ToV8(Mutex<Option<Box<[u8]>>>),
 }
 
-unsafe impl Sync for MagicBuffer {}
-
 impl MagicBuffer {
   pub fn new<'s>(
     scope: &mut v8::HandleScope<'s>,
@@ -36,7 +34,7 @@ impl Clone for MagicBuffer {
 
 impl Default for MagicBuffer {
   fn default() -> Self {
-    MagicBuffer::ToV8(Cell::new(Some(vec![0_u8; 0].into_boxed_slice())))
+    MagicBuffer::ToV8(Mutex::new(Some(vec![0_u8; 0].into_boxed_slice())))
   }
 }
 
@@ -73,7 +71,7 @@ impl DerefMut for MagicBuffer {
 
 impl From<Box<[u8]>> for MagicBuffer {
   fn from(buf: Box<[u8]>) -> Self {
-    MagicBuffer::ToV8(Cell::new(Some(buf)))
+    MagicBuffer::ToV8(Mutex::new(Some(buf)))
   }
 }
 
@@ -100,7 +98,7 @@ impl serde::Serialize for MagicBuffer {
         let value: &[u8] = &*buf;
         value.into()
       }
-      Self::ToV8(x) => x.take().expect("MagicBuffer was empty"),
+      Self::ToV8(x) => x.lock().unwrap().take().expect("MagicBuffer was empty"),
     };
     let hack: [usize; 2] = unsafe { std::mem::transmute(boxed) };
     let f1: u64 = hack[0] as u64;

```

Reply:
Applied